libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Queue/CSFNetwork.m

```diff
@@ -304,11 +302,11 @@ - (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTas
 - (void)receivedDevicedUnauthorizedError:(CSFAction *)action {
 }
 
-#pragma mark - Handling Login State
+#pragma mark - SFAuthenticationManagerDelegate
 
-- (void)userWillLogOutNotification:(NSNotification *)notification {
-    SFUserAccount *userAccount = notification.userInfo[kSFUserAccountKey];
-    [[self class] cleanupNetworkForUserAccount:userAccount];
+- (void)authManager:(SFAuthenticationManager *)manager willLogoutUser:(SFUserAccount *)user {
+    [[self class] removeSharedInstance:user];
```

Comment:
We have taken the exact same approach in SFSmartStore, SFSmartSyncCacheManager, SFSmartSyncMetadataManager, SFSmartSyncSyncManager.

Also, if you were to set some property to self, I don't think it would crash. Once the current instance is dropped from the shared instances dictionary, it doesn't get deallocated right away, my understanding is that it gets an autorelease message (inserted by ARC) and will only be deallocated when the autorelease pool is drained (probably when exiting the method).